client/core/account.go

```diff
@@ -208,24 +209,52 @@ func (c *Core) AccountImport(pw []byte, acct *Account, bonds []*db.Bond) error {
 
 // UpdateCert attempts to connect to a server using a new TLS certificate. If
 // the connection is successful, then the cert in the database is updated.
+// Updating cert for already connected dex will return an error.
 func (c *Core) UpdateCert(host string, cert []byte) error {
-	accountInfo, err := c.db.Account(host)
+	c.connMtx.RLock()
+	dc, found := c.conns[host]
+	c.connMtx.RUnlock()
+	if found && dc.status() == comms.Connected {
+		return errors.New("dex is already connected")
+	}
+
+	acct, err := c.db.Account(host)
 	if err != nil {
 		return err
 	}
 
-	accountInfo.Cert = cert
+	// Stop reconnect retry for previous dex connection first but leave it in
+	// the map so it remains listed incase we need it in the interim.
+	if found {
+		dc.connMaster.Disconnect()
+		dc.acct.lock()
+		dc.booksMtx.Lock()
+		for m, b := range dc.books {
+			b.closeFeeds()
+			if b.closeTimer != nil {
+				b.closeTimer.Stop()
+			}
+			delete(dc.books, m)
+		}
+		dc.booksMtx.Unlock()
+	}
 
-	_, connected := c.connectAccount(accountInfo)
-	if !connected {
-		return errors.New("failed to connect using new cert")
+	acct.Cert = cert
+	dc, err = c.connectDEX(acct)
+	if err != nil {
+		if dc != nil {
+			dc.connMaster.Disconnect() // stop any retry loop for this new connection.
+		}
+		return fmt.Errorf("failed to connect using new cert (will attempt to restore old connection): %v", err)
 	}
 
-	err = c.db.UpdateAccountInfo(accountInfo)
+	err = c.db.UpdateAccountInfo(acct)
 	if err != nil {
 		return fmt.Errorf("failed to update account info: %w", err)
 	}
 
+	c.addDexConnection(dc)
```

Comment:
Between here and `connectDEX` above is where we could consider patching up the `dexAccount` and anything else (which is a lot of maps actually).  But as in our other comment thread, this is a major can of worms.  A different approach might be to patch the previous `dexConnection`'s `WsConn` and `connectionMaster`, but those are meant to be immutable so that sounds much worse.

Reply:
> A different approach might be to patch the previous dexConnection's WsConn and connectionMaster, but those are meant to be immutable so that sounds much worse.

Yeah, thought of this too, but it didn't fit in with the existing design.